### PR TITLE
resolve #186: implement file decompression utility

### LINKED ISSUE #186: Implement file decompression utility
One of the features currently provided by Lutris is file decompression. Nothing too complex is required, a simple utility that can handle the following formats should suffice:

- tar.gz
- zip
- 7z

### PR DESCRIPTION
**Pull request by jthistle**
Hope this is the kind of thing you were looking for! This is a simple script to detect an archive's format and then extract it (optionally into a container directory of your choice). Works with .tar.gz, .zip, and .7z.

Resolves #186

(and apologies for the previous dud PR :))

### CODE REVIEW

File: utils/extract.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    exit 1
+fi
+
+extension="${1#*.}"
+
+case "$extension" in
+    "zip")
+        cmd_needed="unzip"
+        ;;
+    "tar.gz")
```
Comment (Jan200101) [inline utils/extract.sh:29]: this also has an implicit dependency on a gzip implementation

Comment (jthistle) [reply utils/extract.sh:29]: This comes standard with pretty much every Linux distro though, right?

Comment (jthistle) [reply utils/extract.sh:29]: fixed

File: utils/extract.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        echoerr "command '$cmd' not found but is required for file '$1', aborting"
+        exit 1
+    fi
+done
+
+
+# Now we can begin the extraction
+extractdir="$2"
+if [ "$extractdir" == "" ]; then
+    extractdir="."
```
Comment (rockerbacon) [inline utils/extract.sh:52]: I don't see much of a use case for this, I would prefer for the script to exit with an error. Would also remove the need for the check at [ln 67](https://github.com/rockerbacon/lutris-skyrimse-installers/blob/814f5c38e4d568313454ffd0b668dc658d0324f7/utils/extract.sh#L67).

**Review by rockerbacon — CHANGES_REQUESTED** [review]
Looks good even if the comment below is not implemented.

Only thing is that the file was indented with spaces and the rest of the project is indented with tabs. I should probably add an automated check for that.

File: utils/extract.sh (hunk 2 above)
Comment (jthistle) [reply utils/extract.sh:52]: OK, cool. I'll make the second argument required then (and reindent with tabs).

Comment (jthistle) [reply utils/extract.sh]: fixed. In regard to automated checks, you may want to consider running all bash scripts in PRs through `shellcheck`, which is a very useful utility for this kind of work :)

Comment (rockerbacon) [reply utils/extract.sh]: I've used shellcheck in other projects and I'll consider adding it to this one later. But shellcheck doesn't do any style checking, for that you need something like [shfmt](https://github.com/mvdan/sh).

**Comment by rockerbacon** [comment]
Sorry for the late merge.
Thanks a lot for your contribution.

**Pull request merged**